runtime/jvmti/jvmtiHelpers.c

```diff
@@ -1697,3 +1697,23 @@ findDecompileInfo(J9VMThread *currentThread, J9VMThread *targetThread, UDATA dep
 	currentThread->javaVM->walkStackFrames(currentThread, walkState);
 	return (UDATA)walkState->userData1;
 }
+
+void
+ensureHeapWalkable(J9VMThread *currentThread)
+{
+	J9JavaVM *vm = currentThread->javaVM;
+	/* If heap walk is already enabled, nothing need be done */
+	if (J9_ARE_NO_BITS_SET(vm->requiredDebugAttributes, J9VM_DEBUG_ATTRIBUTE_ALLOW_USER_HEAP_WALK)) {
```

Comment:
Can we add an assert that current thread must have exclusive?

Reply:
Done.